repose-aggregator/core/repose-core/src/test/scala/org/openrepose/powerfilter/ReposeRoutingServletTest.scala

```diff
@@ -0,0 +1,230 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.powerfilter
+
+import java.io.IOException
+import java.net.URL
+import java.util.Optional
+
+import javax.servlet._
+import javax.servlet.http.HttpServletResponse._
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.core.LoggerContext
+import org.apache.logging.log4j.test.appender.ListAppender
+import org.junit.runner.RunWith
+import org.mockito.Matchers.{any, anyString, eq => isEq}
+import org.mockito.Mockito.{never, verify, when}
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.io.stream.ReadLimitReachedException
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.core.services.reporting.ReportingService
+import org.openrepose.core.services.reporting.metrics.MetricsService
+import org.openrepose.core.systemmodel.config._
+import org.openrepose.nodeservice.request.RequestHeaderService
+import org.openrepose.nodeservice.response.ResponseHeaderService
+import org.openrepose.powerfilter.ReposeRoutingServletTest._
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfterEach, FunSpec, Matchers}
+import org.springframework.mock.web.{MockHttpServletRequest, MockHttpServletResponse, MockServletConfig}
+
+import scala.collection.JavaConversions._
+
+@RunWith(classOf[JUnitRunner])
+class ReposeRoutingServletTest extends FunSpec with BeforeAndAfterEach with MockitoSugar with Matchers {
+  var mockServletConfig: ServletConfig = _
+  var mockConfigurationService: ConfigurationService = _
+  var mockRequestHeaderService: RequestHeaderService = _
+  var mockResponseHeaderService: ResponseHeaderService = _
+  var mockReportingService: ReportingService = _
+  var mockMetricsService: Optional[MetricsService] = _
+  var reposeRoutingServlet: ReposeRoutingServlet = _
+  var listAppender: ListAppender = _
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+
+    mockServletConfig = new MockServletConfig()
+    mockConfigurationService = mock[ConfigurationService]
+    mockRequestHeaderService = mock[RequestHeaderService]
+    mockResponseHeaderService = mock[ResponseHeaderService]
+    mockReportingService = mock[ReportingService]
+    mockMetricsService = Optional.empty()
+    reposeRoutingServlet = new ReposeRoutingServlet(
+      DefaultClusterId,
+      DefaultNodeId,
+      mockConfigurationService,
+      mockRequestHeaderService,
+      mockResponseHeaderService,
+      mockReportingService,
+      mockMetricsService)
```

Comment:
Looks like this constructor call wasn't updated with the constructor. Test will not compile at the moment.